pkg/storegateway/indexheader/stream_binary_reader.go

```diff
@@ -139,27 +142,27 @@ func newFileStreamBinaryReader(ctx context.Context, binPath string, id ulid.ULID
 	if sparsePersistenceEnabled && r.indexVersion == index.FormatV2 {
 		sparseData, err := os.ReadFile(sparseHeadersPath)
 		if err != nil && !os.IsNotExist(err) {
-			level.Warn(logger).Log("msg", "failed to read sparse index-headers from disk; recreating", "err", err)
+			level.Warn(logger).Log("msg", "failed to read sparse index-headers from disk; recreating", "id", id, "err", err)
```

Comment:
can we wrap the logger at top of `newFileStreamBinaryReader` with the `id` field? that way it will be used in the other debug log in this function and will propagate to the other functions and those don't need to also add the `id` field

Reply:
I tried that originally. It works for the log messages but the key-value pairs set with `log.With(logger, ...)` don't actually make it into the events that are logged as part of the span.